src/cargo/util/progress.rs

```diff
@@ -196,24 +209,23 @@ impl Format {
 
         Some(string)
     }
-}
 
-fn clear(width: usize, config: &Config) {
-    let blank = " ".repeat(width);
-    drop(write!(config.shell().err(), "{}\r", blank));
+    fn width(&self) -> usize {
+        cmp::min(self.max_width, self.max_print)
+    }
 }
 
 impl<'cfg> Drop for State<'cfg> {
     fn drop(&mut self) {
-        clear(self.format.max_width, self.config);
+        self.clear();
     }
 }
 
 #[test]
```

Comment:
Perhaps add a test case where the `max_width` is mutated 😃

Reply:
On the `Format` struct though the methods are all immutable so it may not be testing much :(